report/report-generators/src/main/kotlin/com/testerum/report_generators/reports/remote_server/RemoteServerExecutionListener.kt

```diff
@@ -0,0 +1,55 @@
+package com.testerum.report_generators.reports.remote_server
+
+import com.testerum.common_httpclient.HttpClientService
+import com.testerum.common_httpclient.TesterumHttpClientFactory
+import com.testerum.model.resources.http.request.HttpRequest
+import com.testerum.model.resources.http.request.HttpRequestBody
+import com.testerum.model.resources.http.request.enums.HttpRequestBodyType
+import com.testerum.model.resources.http.request.enums.HttpRequestMethod
+import com.testerum.report_generators.reports.utils.EXECUTION_LISTENERS_OBJECT_MAPPER
+import com.testerum.runner.cmdline.report_type.RunnerReportType
+import com.testerum.runner.cmdline.report_type.builder.EventListenerProperties.JsonEvents.REPORT_SERVER_URL
+import com.testerum.runner.events.execution_listener.ExecutionListener
+import com.testerum.runner.events.model.RunnerEvent
+
+class RemoteServerExecutionListener(private val properties: Map<String, String>) : ExecutionListener {
```

Comment:
Let's add a simple new feature: check if the server is up befor everything (on the first event).
This makes sure the test is not run if the server is down. This is desirable, because if the server is down, we lose all reports.

We can do this by making a call to the ``/health`` endpoint of the remote server. Spring already provides this endpoint as part of the actuator endpoints, but I think we need to enable it (not sure if it's enabled by default). Setting the following property should do it: ``management.endpoint.health.enabled=true``.

Note: it can still happen that the server goes down after we do the check, but we don't care that much. The check will catch the majority of cases.